src/buildExec.ts

```diff
@@ -116,13 +126,23 @@ const buildExec = () => {
     execArgs.push('-g');
   }
   if (gcovArgs) {
-    execArgs.push('--gcovArgs', `${gcovArgs}`);
+    execArgs.push('-gcovArgs', `${gcovArgs}`);
   }
   if (gcovIgnore) {
-    execArgs.push('--gcovIgnore', `${gcovIgnore}`);
+    execArgs.push('-gcovIgnore', `${gcovIgnore}`);
```

Comment:
This change is incorrect and causes files/directories listed in `gcov_ignore` to be included (i.e. the whole option is ignored since version 3.1.2).

Only short CLI option names can be specified with a single dash (`-`) instead of 2 (`--`), see:
https://github.com/codecov/uploader/blob/cb97e5f75b4bba3afed0b07380d5ab29c34d66be/HELPFILE#L40-L42
https://github.com/codecov/uploader/blob/cb97e5f75b4bba3afed0b07380d5ab29c34d66be/src/helpers/cli.ts#L80-L97

So to fix this, one must specify `-gi` or `--gcovIgnore` here.

FWIW, `-gcovArgs`, `-gcovInclude`, `-gcovExecutable`, `-networkFilter`, `-networkPrefix`  are probably having the same issue.

Reply:
> So to fix this, one must specify `-gi` or `--gcovIgnore` here.

Ah, `-gi` is also invalid, since a single dash is only valid for single character aliases, see:
https://github.com/yargs/yargs/issues/1781#issuecomment-708721379

So, this issue affects the following settings:
- `full_report`, uses `-full`, needs to be changed to `--full` or `--fullReport`;
- `gcov_args`, uses `-gcovArgs`, needs to be changed to `--ga` or `--gcovArgs`;
- `gcov_ignore`, uses `-gcovIgnore`, needs to be changed to `--gi` or `--gcovIgnore`;
- `gcov_include`, uses `-gcovInclude`, needs to be changed to `--gI` or `--gcovInclude`;
- `gcov_executable`, uses `-gcovExecutable`, needs to be changed to `--gx` or `--gcovExecutable`;
- `network_filter`, uses `-networkFilter`, needs to be changed to `-i` or `--networkFilter`;
- `networkPrefix`, uses `-networkPrefix`, needs to be changed to `-k` or `--networkPrefix`;
- `swift`, uses `-xs`, needs to be changed to `--xs` or `--swift`;
- `swift_project`, uses `-xsp`, needs to be changed to `--xsp` or `--swiftProject`;
- `xcode`, uses `-xc`, needs to be changed to `--xc` or `--xcode`;
- `xcode`, uses `-xp`, needs to be changed to `--xp` or `--xcodeArchivePath`.

I just opened PR #969 for this.